src/multicorn.c

```diff
@@ -283,11 +283,65 @@ multicornGetForeignRelSize(PlannerInfo *root,
     ListCell   *lc;
     bool		needWholeRow = false;
     TupleDesc	desc;
+    bool        safeToPushLimit = true;
+    Node       *limitNode = root->parse->limitCount;
+    Node       *offsetNode = root->parse->limitOffset;
+    bool        hasOffset = offsetNode != NULL;
 
     baserel->fdw_private = planstate;
     planstate->fdw_instance = getInstance(foreigntableid);
     planstate->foreigntableid = foreigntableid;
 
+    planstate->limit = -1;
+
+    /* Check for joins */
+    if (list_length(root->parse->jointree->fromlist) != 1)
+    {
+        safeToPushLimit = false;
+    }
+
+    /* Check for GROUP BY */
+    if (root->parse->groupClause)
+    {
+        safeToPushLimit = false;
+    }
+
+    /* Check for HAVING clause */
+    if (root->parse->havingQual)
+    {
+        safeToPushLimit = false;
+    }
+
+    /* Check for aggregates */
+    if (root->parse->hasAggs)
+    {
+        safeToPushLimit = false;
+    }
+
+    /* Check for window functions */
+    if (root->parse->hasWindowFuncs)
+    {
+        safeToPushLimit = false;
+    }
+
+    /* Check for ORDER BY */
```

Comment:
This is preventing to push LIMIT when there's an ORDER BY. I suppose if the DAS does implement the required ORDER BY, we should push LIMIT. Is there something more tricky to think about?